src/components/selectionBoxLayer.ts

```diff
@@ -10,29 +11,22 @@ export module Components {
       topLeft: { x: 0, y: 0 },
       bottomRight: { x: 0, y: 0 }
     };
-    private _boxLeftDataValue: number | { valueOf(): number };
-    private _boxRightDataValue: number | { valueOf(): number };
-    private _boxTopDataValue: number | { valueOf(): number };
-    private _boxBottomDataValue: number | { valueOf(): number };
+    private _xExtent: [number | { valueOf(): number }, number | { valueOf(): number }];
```

Comment:
I suggest to type this as 
```typescript
_xExtent: number | { valueOf(): number }[]
```

I agree the current version is better and safer, but it is most definitely going to cause problems on the long run because of incompatibilities. 

For example a linearScale, `_defaultExtent()` returns `number[]` instead of `[number, number]`.

Another example is the hack below
```typescript
protected _setXExtent(xExtent: (number | { valueOf(): number })[]) {
  this._xExtent = [xExtent[0], xExtent[1]];
}
```